crypto/asn1/x_long.c

```diff
@@ -56,11 +56,30 @@ static void long_free(ASN1_VALUE **pval, const ASN1_ITEM *it)
     *(long *)pval = it->size;
 }
 
+static int num_bits_ulong(unsigned long value)
```

Comment:
As already mentioned, sizeof(long)==sizeof(BN_ULONG) is not universally guaranteed, so that there are platforms where this and BN_num_bits_word actually do different things. [And where they do, test_asn1_encode *was* failing.]